testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/AbstractHttpClientTest.java

```diff
@@ -42,12 +42,15 @@
 import java.util.function.Supplier;
 import java.util.stream.IntStream;
 import org.checkerframework.checker.nullness.qual.Nullable;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
 import org.junit.jupiter.api.condition.DisabledIfSystemProperty;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.ValueSource;
 
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
```

Comment:
will this affect subclasses? will that be less obvious compared to making `options` below static?

Reply:
This is a good question, I'll confirm it but I would expect it to, most jupiter functionality goes through hierarchy. I don't think I can use the subclass `configure` pattern (or any pattern I can think of) if `static`, especially because the `@BeforeAll` method must be static if this isn't here.

Reply:
Confirmed that it's respected by subclasses. Though I did come with another pattern using an extension, will play with it in a followup (if I can ever merge past the GwtTest failure :P)